base_classes/NXdata.nxdl.xml

```diff
@@ -60,176 +60,181 @@
+			* the :ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute
+			* :ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;` attributes
 
-		.. code-block::
+		The fields and attributes are defined as follows
 
-			data:NXdata
-			  @signal = "data"
-			  @axes = ["x", "y"]  --> the order matters
-			  data: float[10,20]
-			  x: float[10]        --> coordinates along the first dimension
-			  y: float[20]        --> coordinates along the second dimension
+		1. The :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` fields contain the axis coordinates associated with the signal values.
 
-		In this example each data point ``data[i,j]`` has axis coordinates ``[x[i], y[j]]``.
+		2. The :ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute provides the names of the :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;`
+		   fields to be used as the `default axis` for each dimension of the :ref:`DATA &lt;/NXdata/DATA-field&gt;` fields.
 
-		However, the fields can also have a rank greater than 1, in which case the rank of each
-		:ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` must be equal to the number of data dimensions it spans.
+		3. The :ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;` attributes describe the :ref:`DATA &lt;/NXdata/DATA-field&gt;`
+		   dimensions spanned by the corresponding :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` fields.
 
-		An example of this would be
+		Additional requirements for a complete definition
 
-		.. code-block::
+		1. The length of the :ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute must be equal to the rank of the :ref:`DATA &lt;/NXdata/DATA-field&gt;`
+		   fields. When a particular dimension has no default axis, the string “.” is used in that position.
 
-			data:NXdata
-			  @signal = "data"
-			  @axes = ["x", "y"]  --> the order does NOT matter
-			  @x_indices = [0, 1]
-			  @y_indices = [0, 1]
-			  data: float[10,20]
-			  x: float[10,20]     --> coordinates along both dimensions
-			  y: float[10,20]     --> coordinates along both dimensions
+		2. The number of values in :ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;` must be equal to the rank of the corresponding
+		   :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` field.
 
-		In this example each data point ``data[i,j]`` has axis coordinates ``[x[i,j], y[i,j]]``.
+		3. When :ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;` is missing for a given
+		   :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` field, the positions of the :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;`
+		   field name in the :ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute are used.
 
-		`Dimensions`
+		4. When :ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;` is the same as the indices of "AXISNAME" in the
+		   :ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute, there is no need to provide
+		   :ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;`.
```

Comment:
Only for 1D axis fields?

Reply:
If this applies in general then `@axes=["x", "x"]` would imply `@x_indices=[0,1]`. That would be ok when `x` is 2D.

When `x` is 1D it would not work. It would violate two requirements

* The number of values in AXISNAME_indices must be equal to the rank of the corresponding AXISNAME field.
* The indices of "AXISNAME" in the axes attribute must be a subset of AXISNAME_indices.

So if a 1D `x` needs to work for `@axes=["x", "x"]` we need to change those requirements.

Reply:
Hmm, a pathological case could be `@x_indices=[1,0]`

Reply:
Ok I see. So omitting indices can only work unambiguously in 1D cases.

Reply:
Another pathological case would be `@x_indices=[0,0]`. I'm not sure what this means but technically the definition does not forbid it. So we should say indices cannot contain duplicates?

Reply:
When `x` is 1-D, then `@x_indices=[0,0]` would be compatible with `@axes=["x", "x"]`, right?

Reply:
Why would it not be allowed?

Reply:
In the case of `@x_indices=[0,0]`, this says the same 1-D array applies to both independent axes.

Reply:
Eeeeww!  That's pathological, and likely a mistake.

Reply:
> Eeeeww! That's pathological, and likely a mistake.

Agreed but technically the definition does not present it as far as I can see, which we should probably fix but not sure how.

Stating that `AXISNAME_indices` cannot have duplicates would solve it. But it would prohibit the case where you have for example a single 1D x axis of lets say 100 points and you have a 2D signal of 100 x 100 points and you want x to be used for both dimensions. As I said before, you could define x1 and x2 and both can be symlinks to x so effectively x is used for both dimensions even though `@axes=["x1", "x2"]`.

Reply:
@prjemian Can you modify the code provided in https://github.com/nexusformat/definitions/pull/1396#issuecomment-2380113555 to prove that the current specs allow a 1D signal to be used for multiple dimensions while not breaking support for multi-dimensional axis?

I don't think it is possible.

Reply:
Not sure I know how to proceed on this directly.  For the variety of different formats considered by canSAS, the [discussion](https://wiki.cansas.org/index.php?title=2012_Data_Discussion_Examples) resulted in this [documentation](https://www.cansas.org/canSAS2012/examples.html) where the examples are repeated.  Those are the places I would examine to test multi-dimensional data.